### PR TITLE
Fix Issue Triggering customer invoices when domains have been deleted

### PR DESCRIPTION
**Pull request by biyeun**
https://dimagi-dev.atlassian.net/browse/SAAS-11017

##### SUMMARY
Fixes issue where a customer invoice cannot be created when one of the domains that was part of a particular invoicing period was deleted.

This will unblock Krishna, but I am a bit concerned that we will not legitimately bill for some domains if the DomainUserHistory model was deleted upon domain deletion, but the subscription is left around. It seems like a small enough edge case though for large enterprise projects that maybe that fix doesn't need priority just yet?

##### RISK ASSESSMENT / QA PLAN
Pretty low risk. Fixes bug blogging accounting.

### CODE REVIEW

File: corehq/apps/accounting/invoicing.py [hunk 1]
```diff
@@ -427,6 +427,10 @@ def _email_invoice(self):
 
 
 def should_create_invoice(subscription, domain, invoice_start, invoice_end):
+    from corehq.apps.domain.models import Domain
+    if Domain.get_by_name(domain) is None:
```
Comment (dannyroberts) [inline corehq/apps/accounting/invoicing.py:431]: I think you could make use of [`deleted_domain_exists`](https://github.com/dimagi/commcare-hq/blob/8fb14807fbe323b70fef56d3005588d46954b0d6/corehq/apps/domain/dbaccessors.py#L126-L128) and change this to maybe

```python
if not domain_exists(domain) and deleted_domain_exists(domain):
```

or something like that to just have the condition be super explicit and not vulnerable to any unknown edge cases

File: corehq/apps/accounting/invoicing.py [hunk 2]
```diff
@@ -718,7 +722,12 @@ def quantity(self):
                     history = DomainUserHistory.objects.get(domain=domain, record_date=date)
                     total_users += history.num_users
                 except DomainUserHistory.DoesNotExist:
-                    raise
+                    from corehq.apps.domain.models import Domain
+                    if Domain.get_by_name(domain) is not None:
```
Comment (dannyroberts) [inline corehq/apps/accounting/invoicing.py:726]: same here

**Comment by dannyroberts** [comment]
Looks like some of the tests maybe rely on fake domains that don't exist or something like that

**Comment by biyeun** [comment]
@dannyroberts fixed tests and also added a new test for the deleted domain issue with customer invoices

**Review by dannyroberts — APPROVED** [review]

**Pull request merged**